Night CI 2023-12-26: Failed

action url: https://github.com/spidernet-io/spiderpool/actions/runs/7325759195

```shell
 [Step 6] Install Default-CNI: Calico- Cilium- 
E2E_KUBECONFIG=/home/runner/work/spiderpool/spiderpool/test/.cluster/spiderpool1226015742/.kube/config  \
	E2E_IP_FAMILY=dual \
	E2E_CLUSTER_NAME=spiderpool1226015742 \
	INSTALL_CALICO=true \
	INSTALL_CILIUM=false \
	CALICO_VERSION= \
	DEFAULT_CALICO_VERSION= \
	CALICO_IMAGE_REPO=docker.io \
	CLUSTER_PATH=/home/runner/work/spiderpool/spiderpool/test/.cluster/spiderpool1226015742 \
	CALICO_AUTODETECTION_METHOD="kubernetes-internal-ip" \
	E2E_CILIUM_IMAGE_REPO="quay.io" \
	CILIUM_VERSION= \
	CILIUM_CLUSTER_POD_SUBNET_V4=10.244.64.0/18 \
	CILIUM_CLUSTER_POD_SUBNET_V6=fd00:10:244::/112 \
	DISABLE_KUBE_PROXY=false \
	CALICO_CLUSTER_POD_SUBNET_V4=10.243.64.0/18 \
	CALICO_CLUSTER_POD_SUBNET_V4=10.243.64.0/18 \
	HTTP_PROXY= \
	 bash scripts/install-default-cni.sh
/etc/bash.bashrc: line 7: PS1: unbound variable
install-default-cni.sh : E2E_CLUSTER_NAME spiderpool1226015742 
install-default-cni.sh : INSTALL_CALICO true 
install-default-cni.sh : INSTALL_CILIUM false 
install-default-cni.sh : CLUSTER_PATH /home/runner/work/spiderpool/spiderpool/test/.cluster/spiderpool1226015742 
failed get calico version
make[2]: *** [Makefile:37: kind-init] Error 1
make[2]: Leaving directory '/home/runner/work/spiderpool/spiderpool/test'
make[1]: *** [Makefile:298: e2e_init] Error 2
make[1]: Leaving directory '/home/runner/work/spiderpool/spiderpool'
make: *** [Makefile:330: e2e_init_calico] Error 2
```

@ty-dc failed to get calico version?

--- a/test/Makefile.defs
+++ b/test/Makefile.defs
@@ -134,6 +134,7 @@ E2E_MULTUS_IMAGE_NAME ?= $(E2E_MULTUS_IMAGE_REGISTER)/k8snetworkplumbingwg/multu
 
 #================= calico
 CALICO_VERSION ?=
+DEFAULT_CALICO_VERSION ?= v3.26.4
 
 ifeq ($(E2E_CHINA_IMAGE_REGISTRY),true)
   CALICO_IMAGE_REPO ?= docker.m.daocloud.io

--- a/test/scripts/install-default-cni.sh
+++ b/test/scripts/install-default-cni.sh
@@ -56,9 +56,9 @@ CILIUM_CLUSTER_POD_SUBNET_V6=${CILIUM_CLUSTER_POD_SUBNET_V6:-"fd00:10:244::/112"
 function install_calico() {
     cp ${PROJECT_ROOT_PATH}/test/yamls/calico.yaml $CLUSTER_PATH/calico.yaml
     if [ -z "${CALICO_VERSION}" ]; then
-      [ -n "${HTTP_PROXY}" ] && CALICO_VERSION=$(curl --retry 3 -x "${HTTP_PROXY}" -s https://api.github.com/repos/projectcalico/calico/releases/latest | jq -r '.tag_name')
-      [ -z "${HTTP_PROXY}" ] && CALICO_VERSION=$(curl --retry 3 -s https://api.github.com/repos/projectcalico/calico/releases/latest | jq -r '.tag_name')
-      [ "${CALICO_VERSION}" == "null" ] && { echo "failed get calico version"; exit 1; }
+      [ -n "${HTTP_PROXY}" ] && { CALICO_VERSION_INFO=$(curl --retry 3 --retry-delay 5 -x "${HTTP_PROXY}" -s https://api.github.com/repos/projectcalico/calico/releases/latest); echo ${CALICO_VERSION_INFO}; CALICO_VERSION=$(echo ${CALICO_VERSION_INFO} | jq -r '.tag_name'); }
+      [ -z "${HTTP_PROXY}" ] && { CALICO_VERSION_INFO=$(curl --retry 3 --retry-delay 5 -s https://api.github.com/repos/projectcalico/calico/releases/latest ); echo ${CALICO_VERSION_INFO}; CALICO_VERSION=$(echo ${CALICO_VERSION_INFO} | jq -r '.tag_name'); }
+      [ "${CALICO_VERSION}" == "null" ] && { echo "failed to get the calico version, will try to use default version."; CALICO_VERSION=${DEFAULT_CALICO_VERSION}; }
     else
       CALICO_VERSION=${CALICO_VERSION}
     fi
@@ -83,7 +83,7 @@ function install_calico() {
 
     export KUBECONFIG=${E2E_KUBECONFIG}
     kubectl apply -f  ${CALICO_YAML}
-    sleep 3
+    sleep 10
 
     kubectl wait --for=condition=ready -l k8s-app=calico-node --timeout=${INSTALL_TIME_OUT} pod -n kube-system
     kubectl get po -n kube-system

